skopt/optimizer/optimizer.py

```diff
@@ -400,40 +400,40 @@ def tell(self, x, y, fit=True):
             the optimizer irrespective of the value of `fit`.
         """
         check_x_in_space(x, self.space)
+        self._check_y_is_valid(x, y)
 
+        # take the logarithm of the computation times
         if "ps" in self.acq_func:
             if is_2Dlistlike(x):
-                if np.ndim(y) == 2 and np.shape(y)[1] == 2:
-                    y = [[val, log(t)] for (val, t) in y]
-                    self.Xi.extend(x)
-                    self.yi.extend(y)
-                else:
-                    raise TypeError("expcted y to be a list of (func_val, t)")
+                y = [[val, log(t)] for (val, t) in y]
+            elif is_listlike(x):
+                y = list(y)
+                y[1] = log(y[1])
+
+        return self._tell(x, y, fit=fit)
+
+    def _tell(self, x, y, fit=True):
+        """Same as `tell()` but without checks and without any modification
```

Comment:
Maybe something like:
"""Perform the actual work of incorporating a new point

This method exists to give access to the internals of adding a new point by side stepping the input validation.
"""

It doesn't actually do the same as `tell`, it does all the hard work that `tell` used to do :)